docker-compose.yml

```diff
@@ -177,3 +178,68 @@ services:
   redis:
     image: redis
     restart: always
+
+
+
+
+
+  zookeeper:
+    image: "zookeeper:3.4"
+    container_name: zookeeper
+    ports:
+      - "2181:2181"
+
+  kafka:
+    image: "wurstmeister/kafka:0.10.1.1"
+    container_name: kafka
+    ports:
+      - "9092:9092"
+      - "32773:32773"
+    depends_on:
+      - zookeeper
+    environment:
+      KAFKA_ADVERTISED_HOST_NAME: kafka
+      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock
```

Comment:
The file was mounted to allow kafka to use netcat in order to verify if zookeeper is ready.
"[docker.sock is] the unix socket the Docker daemon listens on by default and it can be used to communicate with the daemon from within a container." (https://medium.com/lucjuggery/about-var-run-docker-sock-3bfd276e12fd)

Kafka will use the 'dump' ZooKeeper command to verify if a certain session with an ephemeral node exists. This command is sent to zookeeper via netcat. (https://zookeeper.apache.org/doc/trunk/zookeeperAdmin.html#sc_zkCommands)

Now that docker-composer uses version 2.1 it might be possible to define a healthcheck on zookeeper instead.

Reply:
on why would one mount /var/run/docker, I'd got the idea - the thing is (quoting the very medium article):

>Note: bind mounting the Docker daemon socket gives a lot of power to a container as it can control the daemon. It must be used with cautious, and only with containers we can trust.

Wong-Strange:
> ... warnings come after the spells.

Also, for zk dump command work (via netcat) it should only needs the name (or at most IP address) of the zookeeper service.

That said, I think using health check might be the way to go.